### PR TITLE
test: integration tests for setting up sign-in methods

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

test: integration tests for setting up sign-in methods

## Testing

Tests passed.

### CODE REVIEW

File: packages/integration-tests/tests/sign-in-experience.test.ts [hunk 1]
```diff
@@ -36,4 +46,37 @@ describe('admin console sign-in experience', () => {
...
+    await updateConnectorConfig(facebookConnectorId, facebookConnectorConfig);
+    await enableConnector(facebookConnectorId);
+
+    // Setup a SMS connector
+    await updateConnectorConfig(twilioSmsConnectorId, twilioSmsConnectorConfig);
+    await enableConnector(twilioSmsConnectorId);
+
+    // Setup an email connector
+    await updateConnectorConfig(sendgridEmailConnectorId, sendgridEmailConnectorConfig);
+    await enableConnector(sendgridEmailConnectorId);
```
Comment (gao-sun) [inline packages/integration-tests/tests/sign-in-experience.test.ts:61]: maybe just two `await Promise.all()`?

File: packages/integration-tests/tests/sign-in-experience.test.ts [hunk 2]
```diff
@@ -36,4 +46,37 @@ describe('admin console sign-in experience', () => {
...
+
+    // Set up sign-in methods
+    const newSignInMethods: Partial<SignInMethods> = {
+      username: SignInMethodState.Primary,
+      sms: SignInMethodState.Secondary,
+      email: SignInMethodState.Secondary,
+      social: SignInMethodState.Secondary,
+    };
+
+    const updatedSignInExperience = await authedAdminApi
```
Comment (gao-sun) [inline packages/integration-tests/tests/sign-in-experience.test.ts:71]: will this API check connector validity?

**Comment by xiaoyijun** [comment]
Convert to draft for this may have concurrency problems on `connectors`.

File: packages/integration-tests/tests/sign-in-experience.test.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/integration-tests/tests/sign-in-experience.test.ts:72]: This API will not, but the connector will be checked on setup.

File: packages/integration-tests/tests/sign-in-experience.test.ts [hunk 3]
```diff
@@ -36,4 +46,45 @@ describe('admin console sign-in experience', () => {
...
+      .json<SignInExperience>();
+
+    expect(updatedSignInExperience.signInMethods).toMatchObject(newSignInMethods);
+
+    // Reset connectors
+    await Promise.all([
+      disableConnector(facebookConnectorId),
+      disableConnector(twilioSmsConnectorId),
+      disableConnector(sendgridEmailConnectorId),
+    ]);
```
Comment (IceHe) [inline packages/integration-tests/tests/sign-in-experience.test.ts:88]: better do it in `afterAll`？

Comment (xiaoyijun) [reply packages/integration-tests/tests/sign-in-experience.test.ts:88]: Connectors are used in this test only; it doesn't make sense to do the reset logic in `afterAll`.

File: packages/integration-tests/src/connectors-mock.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    {
+      subject: 'test-template-subject-value',
+      content: 'Logto: Your passcode is {{code}}. (test template)',
+      usageType: 'Test',
+    },
+  ],
+};
+
+export const sendgridEmailConnectorId = 'sendgrid-email-service';
+export const sendgridEmailConnectorConfig = {
```
Comment (gao-sun) [inline packages/integration-tests/src/connectors-mock.ts:76]: this should using an imported type from the connector actually?

**Review by gao-sun — APPROVED** [review]

File: packages/integration-tests/src/connectors-mock.ts (hunk 4 above)
Comment (xiaoyijun) [reply packages/integration-tests/src/connectors-mock.ts:76]: Add an issue to refactor this: LOG-3829

**Pull request merged**